src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -28,7 +80,13 @@ public static void Run(this IWebHost host)
                     if (!cts.IsCancellationRequested)
                     {
                         Console.WriteLine("Application is shutting down...");
-                        cts.Cancel();
+                        try
+                        {
+                            cts.Cancel();
+                        }
+                        catch (ObjectDisposedException)
```

Comment:
This has always been happening, I only noticed when I ran it under the debugger with all first chance exceptions. The CTS is disposed when the using statement exists on line 107, but it's also captured inside this callback closer the fires on assembly unload (which happens after Main exists).